src/sentry/api/endpoints/group_events.py

```diff
@@ -36,47 +36,71 @@ def get(self, request, group):
         :auth: required
         """
 
-        events = Event.objects.filter(
-            group_id=group.id,
-        )
-
-        query = request.GET.get('query')
-
-        if query:
+        def respond(queryset):
+            return self.paginate(
+                request=request,
+                queryset=queryset,
+                order_by='-datetime',
+                on_results=lambda x: serialize(x, request.user),
+                paginator_cls=DateTimePaginator,
+            )
+
+        events = Event.objects.filter(group_id=group.id)
+
+        try:
+            environment = self._get_environment_from_request(
+                request,
+                group.project.organization_id,
+            )
+        except Environment.DoesNotExist:
+            return respond(events.none())
```

Comment:
couldnt you simply respond with an empty result set in this case?

Reply:
Doesn't look like it:

```
AssertionError at /api/0/issues/11/events/
Expected a `Response`, `HttpResponse` or `HttpStreamingResponse` to be returned from the view, but received a `<class 'django.db.models.query.QuerySet'>`
```